lib/routes/pos/home/pos_invoice.dart

```diff
@@ -103,13 +102,14 @@ class POSInvoiceState extends State<POSInvoice> {
     _invoiceReadyNotificationsSubscription =
         _invoiceBloc.readyInvoicesStream.listen((invoiceReady) {
       // show the simple dialog with 3 states
-      showDialog<bool>(
-        context: context,
-        barrierDismissible: false,
-        builder: (BuildContext context) {
-          return _nfcDialog;
-        }).then((result) {
-        setState(() {
+      if (invoiceReady != null) {
+        showDialog<bool>(
+            context: context,
+            barrierDismissible: false,
+            builder: (BuildContext context) {
+              return _posPaymentDialog;
```

Comment:
I prefer to write here: `return PosPaymentDialog(_invoiceBloc, _posProfileBloc, _scaffoldKey); `
and get rid of the `_posPaymentDialog ` variable.

Reply:
I think that's better too. I just didn't bother to change the old approach but will do now.